python/ray/llm/tests/common/cloud/test_pyarrow_filesystem.py

```diff
@@ -0,0 +1,549 @@
+        mock_pyfs_instance = MagicMock()
+        mock_pyfs.return_value = mock_pyfs_instance
+
+        fs, path = PyArrowFileSystem.get_fs_and_path(
+            "abfss://container@account.dfs.core.windows.net/path/to/file"
+        )
+
+        assert fs == mock_pyfs_instance
+        assert path == "container/path/to/file"
+
+        # Verify the adlfs filesystem was created with correct parameters
+        mock_adlfs.assert_called_once_with(
+            account_name="account", credential=mock_cred.return_value
+        )
+        mock_handler.assert_called_once_with(mock_adlfs_instance)
+        mock_pyfs.assert_called_once_with(mock_handler.return_value)
+
+    @patch("adlfs.AzureBlobFileSystem")
+    @patch("azure.identity.DefaultAzureCredential")
+    @patch("pyarrow.fs.PyFileSystem")
+    @patch("pyarrow.fs.FSSpecHandler")
+    def test_get_fs_and_path_azure(
+        self, mock_handler, mock_pyfs, mock_cred, mock_adlfs
+    ):
+        """Test getting Azure filesystem and path."""
+        mock_adlfs_instance = MagicMock()
+        mock_adlfs.return_value = mock_adlfs_instance
+        mock_pyfs_instance = MagicMock()
+        mock_pyfs.return_value = mock_pyfs_instance
+
+        fs, path = PyArrowFileSystem.get_fs_and_path(
+            "azure://container@account.blob.core.windows.net/path/to/file"
+        )
+
+        assert fs == mock_pyfs_instance
+        assert path == "container/path/to/file"
+
+        # Verify the adlfs filesystem was created with correct parameters
+        mock_adlfs.assert_called_once_with(
+            account_name="account", credential=mock_cred.return_value
+        )
+
+    def test_abfss_uri_validation(self):
+        """Test ABFSS URI validation."""
+        # Test valid URIs
+        valid_uris = [
+            "abfss://container@account.dfs.core.windows.net/path",
+            "abfss://my-container@myaccount.dfs.core.windows.net/deep/nested/path",
+        ]
+
+        for uri in valid_uris:
+            with patch("adlfs.AzureBlobFileSystem"), patch(
+                "azure.identity.DefaultAzureCredential"
+            ), patch("pyarrow.fs.PyFileSystem"), patch("pyarrow.fs.FSSpecHandler"):
+                # Should not raise an exception
+                PyArrowFileSystem._create_abfss_filesystem(uri)
+
+        # Test invalid URIs
+        invalid_uris = [
+            "abfss://container",  # Missing @account
+            "abfss://@account.dfs.core.windows.net/path",  # Empty container
+            "abfss://container@account.wrong.domain/path",  # Wrong domain
+            "abfss://container@.dfs.core.windows.net/path",  # Empty account
+            "abfss://container@account.dfs.core.windows.net",  # No path (but this is actually valid)
+        ]
+
+        for uri in invalid_uris[:-1]:  # Skip the last one as it's actually valid
+            with pytest.raises(ValueError):
+                PyArrowFileSystem._create_abfss_filesystem(uri)
+
+    def test_azure_uri_validation(self):
```

Comment:
I am not so familiar with these azure tests, but make sure you review them if they are not just simple portings from the old test suite.